src/compiler/checker.ts

```diff
@@ -34516,13 +34516,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
     function checkAssertionWorker(node: JSDocTypeAssertion | AssertionExpression, checkMode: CheckMode | undefined) {
         const { type, expression } = getAssertionTypeAndExpression(node);
-        const exprType = checkExpression(expression, checkMode);
         if (isConstTypeReference(type)) {
             if (!isValidConstAssertionArgument(expression)) {
                 error(expression, Diagnostics.A_const_assertions_can_only_be_applied_to_references_to_enum_members_or_string_number_boolean_array_or_object_literals);
             }
-            return getRegularTypeOfLiteralType(exprType);
+            return getRegularTypeOfLiteralType(checkExpression(expression, checkMode));
         }
+        getNodeLinks(node).assertionCheckMode = checkMode;

```

Comment:
I don't think this makes sense because multiple calls can override each other here, right?

Reply:
Yes, you're right. I just wanted to see if having a check mode made any difference whatsoever for `xstate` perf.